internal/provider/serverless_export_resource.go

```diff
@@ -0,0 +1,767 @@
+	ExportOptions *exportOptions `tfsdk:"export_options"`
+	Target        *exportTarget  `tfsdk:"target"`
+	Reason        types.String   `tfsdk:"reason"`
+}
+
+type exportOptions struct {
+	FileType      types.String   `tfsdk:"file_type"`
+	Compression   types.String   `tfsdk:"compression"`
+	Filter        *exportFilter  `tfsdk:"filter"`
+	CsvFormat     *csvFormat     `tfsdk:"csv_format"`
+	ParquetFormat *parquetFormat `tfsdk:"parquet_format"`
+}
+
+type exportFilter struct {
+	Sql   types.String `tfsdk:"sql"`
+	Table *tableFilter `tfsdk:"table"`
+}
+
+type tableFilter struct {
+	Patterns types.List   `tfsdk:"patterns"`
+	Where    types.String `tfsdk:"where"`
+}
+
+type csvFormat struct {
+	Separator  types.String `tfsdk:"separator"`
+	Delimiter  types.String `tfsdk:"delimiter"`
+	NullValue  types.String `tfsdk:"null_value"`
+	SkipHeader types.Bool   `tfsdk:"skip_header"`
+}
+
+type parquetFormat struct {
+	Compression types.String `tfsdk:"compression"`
+}
+
+type exportTarget struct {
+	Type      types.String     `tfsdk:"type"`
+	S3        *s3Target        `tfsdk:"s3"`
+	Gcs       *gcsTarget       `tfsdk:"gcs"`
+	AzureBlob *azureBlobTarget `tfsdk:"azure_blob"`
+}
+
+type s3Target struct {
+	Uri       types.String `tfsdk:"uri"`
+	AuthType  types.String `tfsdk:"auth_type"`
+	AccessKey *accessKey   `tfsdk:"access_key"`
+	RoleArn   types.String `tfsdk:"role_arn"`
+}
+
+type accessKey struct {
+	Id     types.String `tfsdk:"id"`
+	Secret types.String `tfsdk:"secret"`
+}
+
+type gcsTarget struct {
+	Uri               types.String `tfsdk:"uri"`
+	AuthType          types.String `tfsdk:"auth_type"`
+	ServiceAccountKey types.String `tfsdk:"service_account_key"`
+}
+
+type azureBlobTarget struct {
+	Uri      types.String `tfsdk:"uri"`
+	AuthType types.String `tfsdk:"auth_type"`
+	SasToken types.String `tfsdk:"sas_token"`
+}
+type serverlessExportResource struct {
+	provider *tidbcloudProvider
+}
+
+func NewServerlessExportResource() resource.Resource {
+	return &serverlessExportResource{}
+}
+
+func (r *serverlessExportResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_serverless_export"
+}
+
+func (r *serverlessExportResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	var ok bool
+	if r.provider, ok = req.ProviderData.(*tidbcloudProvider); !ok {
+		resp.Diagnostics.AddError("Internal provider error",
+			fmt.Sprintf("Error in Configure: expected %T but got %T", tidbcloudProvider{}, req.ProviderData))
+	}
+}
+
+func (r *serverlessExportResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Serverless Export Resource",
+		Attributes: map[string]schema.Attribute{
+			"export_id": schema.StringAttribute{
```

Comment:
The computed schemas lack UseStateForUnknown() planmodify will increase the diff when apply.